docs/ossm/ossm2-vs-ossm3.md

```diff
@@ -0,0 +1,103 @@
+
+In OpenShift Service Mesh 2.4, a “cluster-wide” mode was introduced to allow a mesh to be cluster-scoped, with the option to limit the mesh using an Istio feature called `discoverySelectors`, which limits the scope of the mesh to a set of namespaces defined with a [label selector](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/). This is aligned with how community Istio worked, and allowed Istio to manage cluster-level resources.
+
+OpenShift Service Mesh 3 converges with Istio in making all meshes “cluster-wide”. This means that Istio control planes are all cluster-scoped resources and the resources `ServiceMeshMemberRoll` and `ServiceMeshMember` are no longer present, with meshes scoped to the entire cluster by default and then scoped down using `discoverySelectors`.
+
+To replace `ServiceMeshMemberRoll`, a `discoverySelector` may be configured to limit the scope of the mesh to one or more namespaces with labels that match the `discoverySelector`'s label selector. 
+
+To replace `ServiceMeshMember`, label(s) may be added to namespaces such that they trigger the label selector specified by `d`iscoverySelector`.
+
+## Multiple Control Planes with Revisions
+
+OpenShift Service Mesh 3 supports multiple service meshes in the same cluster, but in a different manner than in OpenShift Service Mesh 2. A cluster administrator must create multiple `Istio` instances and then configure `discoverySelectors` appropriately to ensure that there is no overlap between mesh namespaces. 
+
+As `Istio` resources are cluster-scoped, they must have unique names to represent unique meshes within the same cluster. The OpenShift Service Mesh 3 operator uses this unique name to create a resource called `IstioRevision` with a name in the format of `{Istio name}` or `{Istio name}-{Istio version}`. Each instance of `IstioRevision` is responsible for managing a single control plane. It does this using Istio's revision labels of the format `istio.io/rev={IstioRevision name}` to identify pods that are part of their control plane. The name with the version identifier becomes important to support canary-style control plane upgrades (more on this in the upgrades section below).
+
+## Sidecar Injection: New Considerations
+
+OpenShift Service Mesh 2 supported using pod annotations and labels to configure sidecar injection and there was no need to indicate which control plane a workload belonged to.
+
+Sidecar injection in OpenShift Service Mesh 3 works the same way as it does for Istio - with pod or namespace labels used to trigger sidecar injection and it may be necessary to include a label that indicates which control plane the workload belongs to.  Note that Istio has deprecated pod annotations in favor of labels for sidecar injection.
+
+When an `Istio` resource has the name “default”, the label `istio-injection=enabled` may be used.
```

Comment:
This restriction will go away once we support the default tag but probably still worth calling out now.

Reply:
Thanks, do you think this page should be updated for the tag section? We can always update the page later...

Reply:
Probably best to leave it for now - though I'll leave this thread open.

Reply:
@dgn Would you be able to say how this section will change once the "tags" feature is implemented or do you think it's best to hold off until that is complete?

Reply:
let's wait until it's actually merged. For the time being, the Istio resource being named `default` is only enough when you're also using InPlace strategy. with RevisionBased updates, at the moment you can't use that label at all

Reply:
Note that this also impacts the doc: https://github.com/openshift-service-mesh/sail-operator/pull/147/files#r1806924107

Reply:
I created OSSM-8285 to update this text (and the injection doc) once the Tags issue (https://github.com/istio-ecosystem/sail-operator/pull/413) is delivered.